internal/attestation/crafter/crafter_test.go

```diff
@@ -487,11 +496,19 @@ func (s *crafterSuite) TestAddMaterialsAutomatic() {
 
 			backend := &casclient.CASBackend{Uploader: uploader}
 
+			if tc.exceedSize {
+				backend.MaxSize = 1
+			}
+
 			c, err := newInitializedCrafter(s.T(), contract, &v1.WorkflowMetadata{}, false, "", runner)
 			require.NoError(s.T(), err)
 
 			kind, err := c.AddMaterialContactFreeAutomatic(context.Background(), "random-id", tc.materialPath, backend, nil)
-			require.NoError(s.T(), err)
+			if err != nil {
+				assert.ErrorIs(s.T(), err, materials.ErrBaseUploadAndCraft)
+			} else {
```

Comment:
this path doesn't seem to test anything. It seems that the tests are missing the expectation on when they should return an error.